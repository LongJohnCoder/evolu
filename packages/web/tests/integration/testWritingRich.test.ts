import { pageDom, pressMany, pageGoto } from './helpers';

beforeEach(async () => {
  await pageGoto('testWritingRich');
});

test('insert text at end', async () => {
  await pressMany('ArrowRight', 7);
  await page.waitFor(50);
  await page.keyboard.press('.');
  await page.waitFor(50);
  await page.keyboard.press(' ');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});

test('insert text at start then middle', async () => {
  await page.keyboard.press('b');
  await page.waitFor(50);
  await page.keyboard.press('l');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});

test('remove text', async () => {
  await page.keyboard.press('b');
  await page.waitFor(50);
  await page.keyboard.press('Backspace');
  await page.waitFor(50);
  await pressMany('ArrowRight', 7);
  await page.waitFor(50);
  await pressMany('Backspace', 7);
  await page.waitFor(50);
  await page.keyboard.press('ArrowDown');
  await page.waitFor(50);
  await page.keyboard.press('ArrowUp');
  await page.waitFor(50);
  await page.keyboard.press('a');
  await page.waitFor(50);
  await page.keyboard.press('b');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});
