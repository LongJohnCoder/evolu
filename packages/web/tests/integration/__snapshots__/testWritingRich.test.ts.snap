// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert text at end 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading. ·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading. ']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·10 {'hasFocus':true,'selection':{'anchor':[0,0,9],'focus':[0,0,9]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading.']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·9 {'hasFocus':true,'selection':{'anchor':[0,0,8],'focus':[0,0,8]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·8 {'hasFocus':true,'selection':{'anchor':[0,0,7],'focus':[0,0,7]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·7 {'hasFocus':true,'selection':{'anchor':[0,0,6],'focus':[0,0,6]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·6 {'hasFocus':true,'selection':{'anchor':[0,0,5],'focus':[0,0,5]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·5 {'hasFocus':true,'selection':{'anchor':[0,0,4],'focus':[0,0,4]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·4 {'hasFocus':true,'selection':{'anchor':[0,0,3],'focus':[0,0,3]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,0,2],'focus':[0,0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0,0],'focus':[0,0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;

exports[`insert text at start then middle 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·blheading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['blheading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,0,2],'focus':[0,0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['bheading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0,0],'focus':[0,0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;

exports[`remove text 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·ab·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['ab']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·21 {'hasFocus':true,'selection':{'anchor':[0,0,2],'focus':[0,0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['a']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·20 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·19 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·18 {'hasFocus':true,'selection':{'anchor':[1,0,0],'focus':[1,0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·17 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['h']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·16 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['he']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·15 {'hasFocus':true,'selection':{'anchor':[0,0,2],'focus':[0,0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['hea']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·14 {'hasFocus':true,'selection':{'anchor':[0,0,3],'focus':[0,0,3]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['head']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·13 {'hasFocus':true,'selection':{'anchor':[0,0,4],'focus':[0,0,4]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['headi']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·12 {'hasFocus':true,'selection':{'anchor':[0,0,5],'focus':[0,0,5]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['headin']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·11 {'hasFocus':true,'selection':{'anchor':[0,0,6],'focus':[0,0,6]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·10 {'hasFocus':true,'selection':{'anchor':[0,0,7],'focus':[0,0,7]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·9 {'hasFocus':true,'selection':{'anchor':[0,0,6],'focus':[0,0,6]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·8 {'hasFocus':true,'selection':{'anchor':[0,0,5],'focus':[0,0,5]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·7 {'hasFocus':true,'selection':{'anchor':[0,0,4],'focus':[0,0,4]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·6 {'hasFocus':true,'selection':{'anchor':[0,0,3],'focus':[0,0,3]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·5 {'hasFocus':true,'selection':{'anchor':[0,0,2],'focus':[0,0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·4 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,0,0],'focus':[0,0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['bheading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,0,1],'focus':[0,0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0,0],'focus':[0,0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':['heading']},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;
